File: nextGreater.py
class Solution:
    # @param A : list of integers
    # @return a list of integers
    def findNext(self, A, i):
        for j in range(i + 1, len(A)):
            if A[j] > A[i]:
                return A[j]
        return -1
        
    def nextGreater(self, A):
        for i in range(len(A)):
            if i == len(A) - 1:
                A[i] = -1
            A[i] = self.findNext(A, i)
        return A

File: nextGreater.java
public int[] nextGreater(int[] A) {
    int[] B = new int[A.length];
    for (int i=0;i<B.length;i++) {
        B[i] = -1;
    }
    for (int i=0; i< A.length-1; i++) {
        int curr = A[i];
        for (int j=i+1; j<A.length; j++) {
            if (A[j] > curr) {
                B[i] = A[j];
                break;
            }
        }
    }
    return B;
}